src/world/scheduler/builder.rs

```diff
@@ -262,6 +262,29 @@ impl WorkloadBuilder {
             Err(err) => panic!("{:?}", err),
         }
     }
+    /// Extends the builder with the following [`WorkloadBuilder`].
+    /// This allows us to collect systems in different builders before joining them together.
+    ///
+    /// [`WorldloadBuilder`]: struct.WorldBuilder.html
+    pub fn extend(&mut self, other: &mut Self) -> &mut Self {
+        let offset_ranges_by = self.borrow_info.len();
+        self.borrow_info.extend(other.borrow_info.drain(..));
+        self.systems.extend(other.systems.drain(..).map(
+            |(type_id, type_name, mut borrow_info_range, is_send_sync, system_fn)| {
+                borrow_info_range.start += offset_ranges_by;
+                borrow_info_range.end += offset_ranges_by;
+                (
+                    type_id,
+                    type_name,
+                    borrow_info_range,
+                    is_send_sync,
+                    system_fn,
+                )
+            },
+        ));
+
+        self
+    }
```

Comment:
```suggestion
    }
    /// Extends the builder with the following [`WorkloadBuilder`].  
    /// This allows us to collect systems in different builders before joining them together.
    ///
    /// [`WorkloadBuilder`]: struct.WorkloadBuilder.html
    pub fn extend(&mut self, mut other: Self) -> &mut Self {
        self.append(&mut other);
        self
    }
```

Tbh I don't know if it is needed. Is there a situation where it could be used where `append` couldn't?

Reply:
I don't think so. I'd sooner someone show a use case for it explicitly than to add more surface area to `shipyard`, but that's just my maintenance opinion, haha.